ticketutil/ticket.py

```diff
@@ -67,19 +105,21 @@ def _create_requests_session(self):
             r.raise_for_status()
             logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
             logging.info("Successfully authenticated to {0}".format(self.ticketing_tool))
+            return s
```

Comment:
I think the reason I did it this way was executing the "raise" in your proposal above resulted in a kind of messy stack trace of the following exceptions:

requests.exception.ConnectionError, requests.packages.urllib3.exceptions.MaxRetryError, requests.packages.urllib3.exceptions.NewConnectionError, ConnectionRefusedError.

By raising the TicketException, we could pass in our own message and the traceback was cleaner:

ticketutil.ticket.TicketException: Error authenticating to http://0.0.0.0:300054/projects.json.

With the new code where I'm verifying the project and ticket_id upon creation of a Ticket object, I do like how the exceptions are cleanly raised in __init__():

```
self.s = self._create_requests_session()
if not self.s:
    raise TicketException("Error authenticating to {0}.".format(self.auth_url))

if not self._verify_project(self.project):
    raise TicketException("Project {0} is not valid.".format(self.project))

if self.ticket_id and not self._verify_ticket_id(self.ticket_id):
    raise TicketException("Ticket {0} is not valid.".format(self.ticket_id))
```
But then again, these exceptions could be immediately raised inside those methods as well, like you're proposing... And I do see your point about how raising it immediately is better. I'll open a PR soon with my code containing the verify_project / verify_ticket_id updates and we can continue the discussion there after you guys take a look.